main.tf

```diff
@@ -0,0 +1,132 @@
+    jenkins_workers_zone                           = "${local.jenkins_workers_project_url}/zones/${var.jenkins_workers_zone}"
+    jenkins_workers_machine_type                   = "${local.jenkins_workers_project_url}/zones/${var.jenkins_workers_zone}/machineTypes/${var.jenkins_workers_machine_type}"
+    jenkins_workers_startup_script                 = "${local.jenkins_workers_startup_script}"
+    jenkins_workers_preemptible                    = "${var.jenkins_workers_preemptible ? "true" : "false"}"
+    jenkins_workers_min_cpu_platform               = "${var.jenkins_workers_min_cpu_platform}"
+    jenkins_workers_labels                         = "${join(",", var.jenkins_workers_labels)}"
+    jenkins_workers_run_as_user                    = "${var.jenkins_workers_run_as_user}"
+    jenkins_workers_boot_disk_type                 = "${local.jenkins_workers_project_url}/zones/${var.jenkins_workers_zone}/diskTypes/${var.jenkins_workers_boot_disk_type}"
+    jenkins_workers_boot_disk_source_image         = "${data.google_compute_image.jenkins_worker.self_link}"
+    jenkins_workers_boot_disk_source_image_project = "${var.jenkins_workers_boot_disk_source_image_project}"
+    jenkins_workers_network                        = "${var.jenkins_workers_network}"
+    jenkins_workers_subnetwork                     = "${var.jenkins_workers_subnetwork}"
+    jenkins_workers_network_tags                   = "${join(",", var.jenkins_workers_network_tags)}"
+    jenkins_workers_service_account_email          = "${var.jenkins_workers_service_account_email}"
+    jenkins_workers_retention_time_minutes         = "${var.jenkins_workers_retention_time_minutes}"
+    jenkins_workers_launch_timeout_seconds         = "${var.jenkins_workers_launch_timeout_seconds}"
+    jenkins_workers_boot_disk_size_gb              = "${var.jenkins_workers_boot_disk_size_gb}"
+    jenkins_workers_num_executors                  = "${var.jenkins_workers_num_executors}"
+    jobs_as_b64_json                               = "${base64encode(jsonencode(var.jenkins_jobs))}"
+  }
+}
+
+resource "google_compute_instance" "jenkins" {
+  project      = "${var.project_id}"
+  name         = "${var.jenkins_instance_name}"
+  machine_type = "${var.jenkins_instance_machine_type}"
+  zone         = "${var.jenkins_instance_zone}"
+
+  tags = "${concat(local.jenkins_tags, var.jenkins_instance_additional_tags)}"
+
+  boot_disk {
+    initialize_params {
+      image = "${data.google_compute_image.jenkins.self_link}"
+    }
+  }
+
+  network_interface {
+    subnetwork         = "${var.jenkins_instance_subnetwork}"
+    subnetwork_project = "${var.project_id}"
+
+    access_config {}
+  }
+
+  metadata = "${merge(local.jenkins_metadata, var.jenkins_instance_additional_metadata)}"
+
+  service_account {
+    email = "${google_service_account.jenkins.email}"
+
+    scopes = [
+      "https://www.googleapis.com/auth/cloud-platform",
+      "https://www.googleapis.com/auth/devstorage.read_write",
```

Comment:
Defer: The [Google Compute Storage Uploader plugin](https://wiki.jenkins.io/display/JENKINS/Google+Cloud+Storage+Plugin) requires `devstorage.full_control` and produces an error in the console, which is mildly annoying and not really critical. It'd be handy to make this a tunable in a later PR. Mentioning this for posterity, should not block the merge.

Resolve this comment if all blockers have been addressed.